### ISSUE TITLE
[BUG] - [514. Freedom Trail] wrong time complexity

### ISSUE THREAD

**Issue by NDR0216** [LeetCode-Feedback/LeetCode-Feedback#22337]
### LeetCode Username

NDR0216

### Problem Number, Title, and Link

514. Freedom Trail https://leetcode.com/problems/freedom-trail/

### Bug Category

Editorial

### Bug Description

I think the time complexity of Approach 4: Shortest Path should be O(R<sup>2</sup>K⋅log(RK))

For each `(keyIndex, ringIndex)` pair, we need to iterate through `characterIndices[ key[keyIndex] ]` to check each occurrence of key[keyIndex], and the size of `characterIndices[ key[keyIndex] ]` can go up to O(R) actually.
For each iteration, we need to push an element to the priority queue, which is O(log(RK)).
That's O(R*log(RK)) for each `(keyIndex, ringIndex)` pair.

The maximum number of pairs we visit is the number of unique possible pairs, which is R*K
Therefore the total time complexity could be O(R<sup>2</sup>K⋅log(RK))

Example:
Suppose the ring consists of only two types of character, AB, and each occurs R/2 times.
For each pair, there are `R/2` occurrences that need to be considered. 
In other words, the time complexity is R/2*O(KR) for each pair. 
That's O(R<sup>2</sup>K⋅log(RK)) for adding up every pair.

### Language Used for Code

None

### Expected behavior

Approach 4: Shortest Path
Time complexity: O(R<sup>2</sup>K⋅log(RK))

**Comment by NDR0216** [comment]
> Although the size of characterIndices[key[keyIndex] ] can reach 
, when we revisit a state, we stop exploration down that path, limiting the total number of edges pushed to the heap.”

But even if we stop exploration when revisiting, the total number of edges in the heap is still O(R2K), right? If there are only O(RK) pairs in the heap, then why would we encounter the same states again?
